### ISSUE TITLE
API documentation

### ISSUE THREAD

**Issue by notramo** [Exafunction/codeium#4]
Is it possible to provide API documentation that would make possible to develop community-supported plugins for less-popular editors?

**Comment by pqn** [comment]
We have some internal documentation for this that we can consider open sourcing, cc @fortenforge. For now, we have a few different open sourced implementations of the API that hopefully can serve as examples.

First party:
* https://github.com/Exafunction/codeium-chrome
* https://github.com/Exafunction/codeium.vim
* https://github.com/Exafunction/codeium.el

Community:
* https://github.com/intitni/CodeiumForXcode

**Comment by notramo** [comment]
@pqn Why is the server executable required? Does it access other files in the project, or provide other statistics? Or it would be possible to write a plugin for Kakoune which would talk directly to the Codeium server API, without running a local process?

**Comment by notramo** [comment]
> We have some internal documentation for this that we can consider open sourcing,

@pqn if it's not something that would give the competition an advantage, then it's definitely worth open sourcing. It would open endless possibilities. For example, I want to create a Codeium plugin for my favourite editor ([Kakoune](https://kakoune.org/)).

**Comment by pqn** [comment]
> @pqn Why is the server executable required? Does it access other files in the project, or provide other statistics? Or it would be possible to write a plugin for Kakoune which would talk directly to the Codeium server API, without running a local process?

Yes, it access other files and maintains state among other things.

> @pqn if it's not something that would give the competition an advantage, then it's definitely worth open sourcing. It would open endless possibilities. For example, I want to create a Codeium plugin for my favourite editor ([Kakoune](https://kakoune.org/)).

It's just a matter of prioritization right now, I believe we intend to do this at some point.

**Comment by hovancik** [comment]
+1 for this. I would love to see Codeium for Pulsar.
